src/fullnode.rs

```diff
@@ -202,6 +200,12 @@ impl Fullnode {
         let exit = Arc::new(AtomicBool::new(false));
         let bank = Arc::new(bank);
 
+        let db_ledger = if db_ledger.is_none() {
```

Comment:
Is Clippy running over this project? It should have suggested:

```rust
let db_ledger = db_ledger.unwrap_or_else(|| Self::make_db_ledger(ledger_path));
```

Reply:
Yep, it is.  When I remove `#[allow(clippy::too_many_arguments)]` in the same file clippy complains.   I've applied this change.